Source/CesiumRuntime/Private/Cesium3DTileset.cpp

```diff
@@ -125,6 +125,79 @@ void ACesium3DTileset::SetMobility(EComponentMobility::Type NewMobility) {
   }
 }
 
+void ACesium3DTileset::SampleHeightMostDetailed(
+    const TArray<FVector>& LongitudeLatitudeHeightArray,
+    FCesiumSampleHeightMostDetailedCallback OnHeightsSampled) {
+  CesiumAsync::Future<Cesium3DTilesSelection::Tileset*> futureNativeTileset =
+      this->GetTileset()
+          ? getAsyncSystem().createResolvedFuture(this->GetTileset())
+          : getAsyncSystem().runInMainThread([this]() {
+              return IsValid(this) ? this->GetTileset() : nullptr;
+            });
+
+  std::vector<CesiumGeospatial::Cartographic> positions;
+  positions.reserve(LongitudeLatitudeHeightArray.Num());
+
+  for (const FVector& position : LongitudeLatitudeHeightArray) {
+    positions.emplace_back(CesiumGeospatial::Cartographic::fromDegrees(
+        position.X,
+        position.Y,
+        position.Z));
+  }
+
+  std::move(futureNativeTileset)
+      .thenImmediately([positions = std::move(positions)](
+                           Cesium3DTilesSelection::Tileset* pTileset) {
+        if (pTileset) {
+          return pTileset->sampleHeightMostDetailed(positions);
+        } else {
+          return getAsyncSystem().createResolvedFuture(
+              Cesium3DTilesSelection::SampleHeightResult{
+                  {},
+                  {},
+                  {"Could not sample heights from tileset because it has not been created."}});
+        }
+      })
+      .thenInMainThread(
+          [this, OnHeightsSampled = std::move(OnHeightsSampled)](
```

Comment:
`std::forward` should only used with forwarding references. A forwarding reference looks like `T&&`, where `T` can be any template parameter name, but it has to be _just_ a template parameter (not some expression involving a template parameter).

In this case, we have `FCesiumSampleHeightMostDetailedCallback OnHeightsSampled`, so it's passed by value (not a reference at all). We then use `std::move` to move this instance into the lambda, rather than copying it.

`OnHeightsSampled` is passed by value instead of r-value reference so that it can be called with either an r-value or an l-value.

Reply:
Ah that makes sense now. Thanks for explaining 😄